src/StepperISR_due.cpp

```diff
@@ -1,401 +1,691 @@
+      break;
+    case 4:
+      Serial.print("Attach 4 to ");
+      Serial.println(ISRQueueNameStr(4));
+      attachInterrupt(digitalPinToInterrupt(step_pin), ISRQueueName(4), RISING);
+      break;
+    case 5:
+      Serial.print("Attach 5 to ");
+      Serial.println(ISRQueueNameStr(5));
+      attachInterrupt(digitalPinToInterrupt(step_pin), ISRQueueName(5), RISING);
+      break;
+  }
+
+  digitalWrite(step_pin, LOW);
+  pinMode(step_pin, OUTPUT);
+  NVIC_SetPriority(PIOA_IRQn, 0);
+  NVIC_SetPriority(PIOB_IRQn, 0);
+  NVIC_SetPriority(PIOC_IRQn, 0);
+  NVIC_SetPriority(PIOD_IRQn, 0);
+  PWM_INTERFACE->PWM_IDR2 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_IDR1 = 0x00FFFF0F;
+  NVIC_EnableIRQ(PWM_IRQn);
+  PWM_INTERFACE->PWM_IDR2 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_IDR1 = 0x00FFFF0F;
+
+  connect();
+}
+
+void StepperQueue::connect() {
+  PIO_Configure(g_APinDescription[_step_pin].pPort,
+                g_APinDescription[_step_pin].ulPinType,
+                g_APinDescription[_step_pin].ulPin,
+                g_APinDescription[_step_pin].ulPinConfiguration);
+  PWMC_ConfigureChannel(PWM_INTERFACE, mapping->channel, PWM_CMR_CPRE_CLKB, 0,
+                        0);
+  // 21 pulses makes for a microsecond pulse.  I believe my drivers need 5us.
+  // I'll set this to 10 just to make sure the drivers receive this.  At 20us
+  // pulse speed, this is still a 1/2 duty cycle PWM.  Should be easy for any
+  // driver to pickup.
+
+  PWM_INTERFACE->PWM_IDR2 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_IDR1 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_DIS = PWM_INTERFACE->PWM_DIS & (~mapping->channelMask);
+  PWMC_SetPeriod(PWM_INTERFACE, mapping->channel, 65535);
+  PWM_INTERFACE->PWM_DIS = PWM_INTERFACE->PWM_DIS & (~mapping->channelMask);
+  PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CDTY = 21 * 10;
+  PWM_INTERFACE->PWM_DIS = PWM_INTERFACE->PWM_DIS & (~mapping->channelMask);
+  PWM_INTERFACE->PWM_IDR2 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_IDR1 = 0x00FFFF0F;
+
+  // Rising edge, so we get the interrupt at the beginning of the pulse (gives
+  // us some extra time) and so we only get 1 int/pulse Change would give 2.
+  switch (_queue_num) {
+    case 0:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(0),
+                      RISING);
+      break;
+    case 1:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(1),
+                      RISING);
+      break;
+    case 2:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(2),
+                      RISING);
+      break;
+    case 3:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(3),
+                      RISING);
+      break;
+    case 4:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(4),
+                      RISING);
+      break;
+    case 5:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(5),
+                      RISING);
+      break;
+  }
+  _connected = true;
```

Comment:
Can replace the above switch with this one liner ?:
```
attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(_queue_num), RISING);
```

Reply:
yes, it can be.  The reason it exists the way it does is from when I thought the AtmelICE would be useful.  I reorganized code to not use variables in some places where speed didn't matter to make it clear what the variable was.  Even that didn't work so well.  I thought I removed all of that code.  You can see in there I even ended up resorting to Serial.prints despite trying to use the hardware JTAG debugger!  Eventually I did give up on the debugger.  The hardware is still around me somewhere....but I'm not even sure where :)  I'll make that change and test it as well.

Reply:
Ah, nope, I did try to remove it previously.  Its because its generating a function name, so it needs to paste in the value passed.  passing queue_num results in a function called Queuequeue_numISR which obviously doesn't exist ;)  cant mix variables with preprocessor directives.  I knew that, and tried it once already, and now twice!

Reply:
sorry for the inconvenience. I will later check, how/if platformio can be configured to build the examples for due, too. Then this just comes out for free from github actions’ CI